integreat_compass/cms/views/offers/offer_form_view.py

```diff
@@ -0,0 +1,158 @@
+import logging
+
+from django.contrib import messages
+from django.db import transaction
+from django.shortcuts import render
+from django.utils.decorators import method_decorator
+from django.utils.translation import gettext as _
+from django.views.generic import TemplateView
+
+from ...decorators import permission_required
+from ...forms import (
+    ContactForm,
+    LocationForm,
+    OfferForm,
+    OfferVersionForm,
+    OrganizationForm,
+)
+from ...models import Offer
+
+logger = logging.getLogger(__name__)
+
+
+@method_decorator(permission_required("cms.edit_offer"), name="dispatch")
+class OfferFormView(TemplateView):
+    """
+    View for the offer form
+    """
+
+    template_name = "offers/offer_form.html"
+
+    def get(self, request, *args, **kwargs):
+        r"""
+        Render :class:`~integreat_compass.cms.forms.offers.contact_form.ContactForm`,
+               :class:`~integreat_compass.cms.forms.offers.location_form.LocationForm`,
+               :class:`~integreat_compass.cms.forms.offers.offer_form.OfferForm`,
+               :class:`~integreat_compass.cms.forms.offers.offer_version_form.OfferVersionForm` and
+               :class:`~integreat_compass.cms.forms.offers.organization_form.OrganizationForm`
+
+        :param request: The current request
+        :type request: ~django.http.HttpRequest
+
+        :param \*args: The supplied arguments
+        :type \*args: list
+
+        :param \**kwargs: The supplied keyword arguments
+        :type \**kwargs: dict
+
+        :raises ~django.core.exceptions.PermissionDenied: If user does not have the permission to edit the specific page
+
+        :return: The rendered template response
+        :rtype: ~django.template.response.TemplateResponse
+        """
+
+        offer = Offer.objects.filter(id=kwargs.get("pk")).first()
+        if offer is None and kwargs.get("pk") is not None:
+            raise Offer.DoesNotExist
+
+        (offer_version, offer_contact, offer_location, offer_organization) = (
+            (
+                offer.versions.first(),
+                offer.offer_contact,
+                offer.location,
+                offer.organization,
+            )
+            if offer is not None
+            else (None, None, None, None)
+        )
```

Comment:
```suggestion
        if offer_id := kwargs.get("pk"):
            offer = Offer.objects.get(id=offer_id)
            offer_version = offer.versions.first()
            offer_contact = offer.offer_contact
            offer_location = offer.location
            offer_organization = offer.organization
        else:
            offer = offer_version = offer_contact = offer_location = offer_organization = None  # fmt: skip
```
Maybe this would be more concise?

Reply:
Is i? 😅 But sure, I do not mind either way